src/util/data.js

```diff
@@ -238,18 +238,45 @@ export const ensureAvailabilityException = availabilityException => {
 };
 
 /**
- * Get the display name of the given user. This function handles
+ * Get the display name of the given user as string. This function handles
  * missing data (e.g. when the user object is still being downloaded),
  * fully loaded users, as well as banned users.
  *
- * For banned users, a translated name should be provided.
+ * For banned or deleted users, a translated name should be provided.
+ *
+ * @param {propTypes.user} user
+ * @param {String} defaultUserDisplayName
+ *
+ * @return {String} display name that can be rendered in the UI
+ */
+export const userDisplayNameAsString = (user, defaultUserDisplayName) => {
+  const hasAttributes = user && user.attributes;
+  const isBanned = hasAttributes && user.attributes.banned;
+  const isDeleted = hasAttributes && user.attributes.deleted;
+  const hasProfile = hasAttributes && user.attributes.profile;
+
+  if (isBanned || isDeleted) {
+    return defaultUserDisplayName;
+  } else if (hasProfile) {
+    return user.attributes.profile.displayName;
+  } else {
+    return '';
+  }
+};
```

Comment:
I'd suggest that this is turned the other way around:
```js
if (hasProfile) {
  return user.attributes.profile.displayName;
} else {
  return defaultUserDisplayName || '';
}
```

Then there's no need to check if the user is banned or deleted or anything else.